### PR TITLE
Fix/riot exception

### PR DESCRIPTION
**Pull request by PhilippKuntschik**

### CODE REVIEW

File: src/test/java/com/weblyzard/sparql/StreamingResultSetTest.java [hunk 1]
```diff
@@ -106,6 +105,7 @@ public void testInvalidData() throws IOException {
         try (StreamingResultSet s = new StreamingResultSet(bufferedReader)) {
             @SuppressWarnings("unused")
             List<Map<String, Node>> result = Lists.newArrayList(s);
```
Comment (AlbertWeichselbraun) [inline src/test/java/com/weblyzard/sparql/StreamingResultSetTest.java]: I don't think that the SupressWarnings should be necessary any more.

File: src/main/java/com/weblyzard/sparql/StreamingResultSet.java [hunk 2]
```diff
@@ -162,4 +162,9 @@ public int getRowNumber() {
     public void close() throws IOException {
         in.close();
     }
+    
+    private static String escape(String in) {
+        String result = in.replaceAll("\"\"", "\\\\\\\"");
+        return result;
```
Comment (AlbertWeichselbraun) [inline src/main/java/com/weblyzard/sparql/StreamingResultSet.java]: can you please add a pointer to the documentation suggesting to escape nodes using this schema.

File: pom.xml [hunk 3]
```diff
@@ -267,16 +255,42 @@
 			<scope>test</scope>
 		</dependency>
 
+		<!-- JAXB dependencies required from java9 onwards -->
```
Comment (AlbertWeichselbraun) [inline pom.xml]: are you sure that we need these imports (the library does not provide any jaxb services).

File: pom.xml [hunk 4]
```diff
@@ -108,63 +109,50 @@
...
+				<executions>
+					<execution>
+						<id>attach-sources</id>
+						<goals>
+							<goal>jar-no-fork</goal>
+						</goals>
+					</execution>
+				</executions>
+			</plugin>
+	<!-- <plugin> <groupId>org.apache.maven.plugins</groupId> <artifactId>maven-javadoc-plugin</artifactId>
```
Comment (AlbertWeichselbraun) [inline pom.xml]: doesn't this break the generation of the javadocs?

**Review by AlbertWeichselbraun — COMMENTED** [review]
hi phil

thank you for this merge request.

i am not sure whether it is necessary to change all libraries to java 10 (especially if this does not yield any improvements in efficiency).

can you please address the points outlined in the review.

many thanks,
  albert :)

File: pom.xml (hunk 3 above)
Comment (PhilippKuntschik) [reply pom.xml]: these dependencies are not used by the project, but restrain jersey to throw warnings about classes that were deprecated with java 9. jersey is a dependency included by junit-docker-rule.

as a solution, I set those dependencies to scope test, which also fixes the warnings.

**Pull request merged**